src/evaluate.js

```diff
@@ -15,83 +15,94 @@ export default function evaluate(tokens, expr, values) {
     var item = tokens[i];
     var type = item.type;
     if (type === INUMBER || type === IVARNAME) {
-      nstack.push(item.value);
+      nstack.push(item.value, item.value);
     } else if (type === IOP2) {
-      n2 = nstack.pop();
-      n1 = nstack.pop();
+      var right = nstack.pop();
+      var left = nstack.pop();
+      n2 = right.value;
+      n1 = left.value;
```

Comment:
stupid question, is your pushing item.value, when you pop it, you need to .value it again to set on n1 and n2? :)
also i can see that youre using popValue instead of pop. Is it applicable here? :)

Reply:
Well, in this part we need both the value and the context around it.
Each time the context is pushed to the stack, it contains both the value and the related token. However, most of the time, only the value is needed, hence the `.popValue()`. For the percentage operator, We need to know what was the token that is linked to the value in order to infer what formula to apply.

Reply:
aaah yes, i see it later you need the token from the same variable :), nice job!